gui/mainmenu.py

```diff
@@ -49,35 +48,48 @@ def __init__(self, widget, ic, ienv):
         else:
             self.actionSearch.triggered.connect(self.search)
             self.actionSaveConfig.triggered.connect(self.saveConfig)
-            # Browser tab needed for Search
+            # self.actionExportMetadata.triggered.connect(self.exportMeta)
+            # needed for Search
             self.irodsBrowser = gui.IrodsBrowser.IrodsBrowser(ic)
-            self.tabWidget.addTab(self.irodsBrowser, 'Browser')
-            # Optional tabs
-            if ('ui_tabs' in ienv) and (ienv['ui_tabs'] != []):
+            self.tabWidget.addTab(self.irodsBrowser, "Browser")
+            ui_tabs_lookup = {
+                "tabUpDownload": self.setupTabUpDownload,
+                "tabELNData": self.setupTabELNData,
+                "tabDataBundle": self.setupTabDataBundle,
+                "tabCreateTicket": self.setupTabCreateTicket,
+            }
+            if ("ui_tabs" in ienv) and (ienv["ui_tabs"] != ""):
                 # Setup up/download tab, index 1
-                if 'tabUpDownload' in ienv['ui_tabs']:
-                    self.updownload = gui.IrodsUpDownload.IrodsUpDownload(
-                        ic, self.ienv)
-                    self.tabWidget.addTab(self.updownload, "Data Transfers")
-                    log_handler = QPlainTextEditLogger(self.updownload.logs)
-                    logging.getLogger().addHandler(log_handler)
-                # Elabjournal tab, index 2
-                if 'tabELNData' in ienv['ui_tabs']:
-                    self.elnTab = gui.elabUpload.elabUpload(ic)
-                    self.tabWidget.addTab(self.elnTab, "ELN Data upload")
-                # Data (un)bundle tab, index 3
-                if 'tabDataBundle' in ienv['ui_tabs']:
-                    self.bundleTab = gui.IrodsDataBundle.IrodsDataBundle(ic, self.ienv)
-                    self.tabWidget.addTab(self.bundleTab, "(Un)Bundle data")
-                # Grant access by tickets, index 4
-                if 'tabCreateTicket' in ienv['ui_tabs']:
-                    self.createTicket = gui.irodsCreateTicket.irodsCreateTicket(ic, self.ienv)
-                    self.tabWidget.addTab(self.createTicket, "Create access tokens")
-            # General info
+                for tab in ienv["ui_tabs"]:
+                    if tab in ui_tabs_lookup:
+                        ui_tabs_lookup[tab](ic)
+                    else:
+                        logging.error("Unknown tab \"{uitab}\" defined in irods environment file".format(uitab=tab))
```

Comment:
Nice!  I had not thought to do this, but it is the kind of consolidation method I would have chosen.